src/server.py

```diff
@@ -605,5 +588,105 @@ def template(filename):
         return response_body
     abort(404)
 
+
+def update_preseed():
+    """ we sync preseed.json to jobpririties in server on startup, since that is
+        the only time we expect preseed.json to change. """
+
+    # get preseed data first
+    preseed_path = os.path.join(os.path.dirname(SCRIPT_DIR), 'src', 'preseed.json')
+    preseed = []
+    with open(preseed_path, 'r') as fHandle:
+        preseed = json.load(fHandle)
+
+    # Preseed data will have fields: buildtype,testtype,platform,priority,timeout,expires
+    # The expires field defaults to 2 weeks on a new job in the database
+    # Expires field has a date "YYYY-MM-DD", but can have "*" to indicate never
+    # Typical priority will be 1, but if we want to force coalescing we can do that
+    # One hack is that if we have a * in a buildtype,testtype,platform field, then
+    # we assume it is for all flavors of the * field: i.e. linux64,pgo,* - all tests
+    # assumption - preseed fields are sanitized already - move parse_testtype to utils.py ?
+    for job in preseed:
+        _buildsystem = "buildbot" # use this as a default
```

Comment:
I decided that '*' would represent all.  This could be problematic if we have a third system, but right now we have just 2: buildbot and taskcluster.

You ask a good question about why this is defaulted to "buildbot" and what happens with taskcluster jobs.  I had to re-read the code a few times to get this.

In this function we are specifically dealing with new preseed data.  Right now preseed is primarily talos which is all buildbot.  So the idea is we would add a new job that doesn't exist in with a default buildsystem.  If the job exists already, then we compare the data and make sure that we support both buildsystems with a "*".

I have a bug in here, and I believe in preseed.json we need to add "buildsystem" to the fields.  This would allow us to not hardcode data and make things much more accurate!

Thanks for pointing this out.